### PR TITLE
Revive command line tests (berkeley)

### PR DESCRIPTION
**Pull request by dkijania**
Explain your changes:

I took a liberty to revive command_line_tests. They really looked promising as a cornerstone for single node local tests. 
I've maintained them enough to be green, migrated to alcotest and finally refactored in a way that adding new tests cases should be easy. New test cases has assumption that they are operating on single node. Added new pipeline for them

Explain how you tested your changes:

Created separate pipeline for above tests

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

File: src/test/command_line_tests/runner.ml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      ~finally:(fun () ->
+        ( if !test_failed then
+          let contents =
+            Core.In_channel.(
+              with_file (ConfigDirs.mina_log config.dirs) ~f:input_all)
+          in
+          printf
+            !"**** DAEMON CRASHED (OUTPUT BELOW) ****\n%s\n************\n%!"
+            contents ) ;
+        if clean_up then remove_config_dirs (ConfigDirs.dirs config.dirs)
```
Comment (deepthiskumar) [inline src/test/command_line_tests/runner.ml:30]: When is this true?

File: src/test/command_line_tests/mina_daemon.ml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+open Config
+open Core
+open Async
+
+module MinaDaemon = struct
+  type t = { process : Process.t; config : Config.t }
+
+  let create process config = { process; config }
+
+  let kill t =
```
Comment (deepthiskumar) [inline src/test/command_line_tests/mina_daemon.ml:10]: more like die or crash :)

**Review by deepthiskumar — APPROVED** [review]
These were running as part of dev unit tests, weren't they? 
But this looks great! I like how this is structured. Makes it easier to add more cases.

**Comment by deepthiskumar** [comment]
Could you create a PR against berkeley? Any tests that doesn't impact production code can/should go into berkeley

File: src/test/command_line_tests/runner.ml (hunk 1 above)
Comment (dkijania) [reply src/test/command_line_tests/runner.ml:30]: added clean_up to config struct

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-build-me

**Review by shimkiv — APPROVED** [review]

**Comment by dkijania** [comment]
!ci-build-me

File: src/test/command_line_tests/command_line_tests.ml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      let%map _ = MinaCli.stop_daemon mina_cli in
+      ()
+    in
+    Async.Thread_safe.block_on_async_exn (fun () -> TestRunner.run test_case)
+end
+
+let () =
+  let open Alcotest in
+  run "Test commadline."
+    [ ( "commadnline-tests"
```
Comment (psteckler) [inline src/test/command_line_tests/command_line_tests.ml:44]: typo here

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**